src/controller/CHIPPersistentStorageDelegate.h

```diff
@@ -0,0 +1,64 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <core/CHIPCore.h>
+#include <support/DLLUtil.h>
+
+namespace chip {
+namespace DeviceController {
+
+class DLL_EXPORT PersistentStorageDelegate
```

Comment:
Should this be in `controller` or someplace more general? How does the controllee store its keys?

Are we considering the `ConfigurationManager` code deprecated?

Reply:
It used to be in `DeviceController` header file. As per @andy31415 comment, I moved it to a separate file.
`ConfigurationManager` is not deprecated. This is specifically for controllee applications (which may not implement the full platform layer, e.g. iOS, Android). For embedded `controllee` apps, this interface can be implemented on top of configuration manager.